### PR TITLE
Add the function of setting default args. Rebuild the code.

### PR DESCRIPTION
**Pull request by Anti-Li**
1. Add the function of setting default run and compilation args required by #13 and #14. It can be set in Settings now.
2. Rebuild the code. Merge the case "cpp" and the case "c". Rebuild the class "Settings". Add class "Args" which can help to do args operation.

### CODE REVIEW

**Comment by danielpinto8zz6** [comment]
Nice, Ill review and merge when I get home.
Thank you

**Comment by danielpinto8zz6** [comment]
Btw you included some unneded and old commits

**Comment by Anti-Li** [comment]
How to delete them?

**Comment by danielpinto8zz6** [comment]
Maybe it happend because you didnt rebased the code. 
No problem, I cleanup

**Comment by danielpinto8zz6** [comment]
Log from console : 
`rejected promise not handled within 1 second
`

**Comment by danielpinto8zz6** [comment]
Also, instead of the args class wouldn't be simpler to use a string and split it like I did before?

**Comment by Anti-Li** [comment]
So how to rebase the code?

**Comment by danielpinto8zz6** [comment]
better way to clone this repo again and start working from there

**Comment by danielpinto8zz6** [comment]
check it : https://github.com/danielpinto8zz6/c-cpp-compile-run/commit/574f568a4957c7582a6a744555a0cb67b40123d0
isn't that simpler and better?

**Comment by Anti-Li** [comment]
You are right.
I originally thought that if we have base args "gcc a.cpp -o a.exe -lstdc++" and "-lstdc++ -std=c++17" from user, using string but not set, the command will be "gcc a.cpp -o a.exe -lstdc++ -lstdc++ -std=c++17", which will cause a bug from g++.
When I hear your thoughts, I test in g++, found it will not cause a bug. So you can simplify it.

**Comment by Anti-Li** [comment]
There is some unneded and old commits because last time you did not merge my pull request but merge yours into mine, it's different, I think.

**Comment by Anti-Li** [comment]
If your think your method to realize is better than mine, you can edit the pull request but not merge your code, I think.

**Comment by Anti-Li** [comment]
> better way to clone this repo again and start working from there

I never heard this way. Maybe I'm ignorant? 😂

File: src/CompileRun.ts [hunk 1]
```diff
@@ -23,78 +56,71 @@ export class CompileRun {
 
         let currentFileName = currentFile.fileName;
 
-        if (Settings.saveBeforeCompile) {
+        if (Settings.saveBeforeCompile()) {
```
Comment (Anti-Li) [inline src/CompileRun.ts]: Like this, useful changed, really influences, I think.

**Comment by danielpinto8zz6** [comment]
Well, its your commit, I just edited some things to fit better in code and look more clean.
No ahah

**Comment by Anti-Li** [comment]
image
You can change my pull request. And you can change it and then merge it.

**Comment by danielpinto8zz6** [comment]
https://github.com/danielpinto8zz6/c-cpp-compile-run/commits/master

Check it, I made a few changes

**Comment by danielpinto8zz6** [comment]
btw you know why the promise error in your code?

**Comment by Anti-Li** [comment]
But do you know merging the pull request is different from merging your repo into mine? 😂

**Comment by Anti-Li** [comment]
This might be a bug, I'm trying to find why. :scream: But the extension can work perfectly.

**Comment by Anti-Li** [comment]
But now you choose the second way, and every time my pull request include old commit, though it's already "merge". So please merge the pull request, and then use the better method to realize in your repo. That's the standard way, I think. 😄 (When there's no bug)

**Comment by danielpinto8zz6** [comment]
No, it happens because you don't keep your repo synced with that one, so, when you pr a lot of merges appear because your code wasn't synced.

**Comment by Anti-Li** [comment]
I don't think so. Look that.
image
Last time you didn't merge it, but merging your repo into mine.(If you merge it, the image will be purple) I think it's the real reason why here's old commit my old pull request included.

**Comment by danielpinto8zz6** [comment]
I still cherry picked your commits and authored you, the commits are done,
but to avoid that make sure you clone this repo again and do `git pull` every time before you make any change

**Comment by Anti-Li** [comment]
Maybe you are right, maybe what you say is a different way.
But let's try okay? Try.
> Practice is the sole criterion for testing truth.
Let me make a test repo in my repos to try, please make some changes and make a pull request to simulate what I did in this repo.

**Comment by Anti-Li** [comment]
https://github.com/Anti-Li/TEST

**Comment by danielpinto8zz6** [comment]
https://github.com/Anti-Li/TEST/pull/1

**Comment by Anti-Li** [comment]
> Okay. You are right, I know less about Git.
> 
> https://git-scm.com/book/en/v2/Git-Branching-Rebasing
> https://www.git-tower.com/learn/git/ebook/cn/command-line/advanced-topics/rebase
> 
> They said merge and rebase are two different way but both can integrate changes from one branch into another.
> 
> > Rebase vs. Merge
> > Now that you’ve seen rebasing and merging in action, you may be wondering which one is better. Before we can answer this, let’s step back a bit and talk about what history means.
> > One point of view on this is that your repository’s commit history is a record of what actually happened. It’s a historical document, valuable in its own right, and shouldn’t be tampered with. From this angle, changing the commit history is almost blasphemous; you’re lying about what actually transpired. So what if there was a messy series of merge commits? That’s how it happened, and the repository should preserve that for posterity.
> > The opposing point of view is that the commit history is the story of how your project was made. You wouldn’t publish the first draft of a book, and the manual for how to maintain your software deserves careful editing. This is the camp that uses tools like rebase and filter-branch to tell the story in the way that’s best for future readers.
> 
> Thank you. I know the new way to integrate changes from one branch into another now.

Happy cooperation! I will try to 

> I still cherry picked your commits and authored you, the commits are done,
> but to avoid that make sure you clone this repo again and do `git pull` every time before you make any change

next time.

**Pull request closed without merging**